File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/web/EcosystemNativeApi.java
package com.kin.ecosystem.web;

import android.webkit.JavascriptInterface;
import com.kin.ecosystem.Log;
import com.kin.ecosystem.Logger;

class EcosystemNativeApi {

    private static final String TAG = EcosystemNativeApi.class.getSimpleName();

    private EcosystemWebPageListener listener;

    @JavascriptInterface
    public void loaded() {
        Logger.log(new Log().withTag(TAG).text("loaded()"));
        if (listener != null) {
            listener.onPageLoaded();
        }
    }

    @JavascriptInterface
    public void handleCancel() {
        Logger.log(new Log().withTag(TAG).text("handleCancel()"));
        if (listener != null) {
            listener.onPageCancel();
        }
    }

    @JavascriptInterface
    public void handleResult(final String result) {
        Logger.log(new Log().withTag(TAG).text("handleResult(\"" + result + "\")"));
        if (listener != null) {
            listener.onPageResult(result);
        }
    }

    @JavascriptInterface
    public void displayTopBar(boolean shouldDisplay) {
        Logger.log(new Log().withTag(TAG).text("displayTopBar(\"" + shouldDisplay + "\")"));
        if (listener != null) {
            if (shouldDisplay) {
                listener.showToolbar();
            } else {
                listener.hideToolbar();
            }
        }
    }

    @JavascriptInterface
    public void handleClose() {
        Logger.log(new Log().withTag(TAG).text("handleClose()"));
        if (listener != null) {
            listener.onPageClosed();
        }
    }

    void setListener(final EcosystemWebPageListener listener) {
        this.listener = listener;
    }
}

File: sdk/src/main/java/com/kin/ecosystem/web/EcosystemNativeApi.kt
package com.kin.ecosystem.web

import android.webkit.JavascriptInterface
import com.kin.ecosystem.core.Log
import com.kin.ecosystem.core.Logger

internal class EcosystemNativeApi {

    private var listener: EcosystemWebPageListener? = null

    @JavascriptInterface
    fun loaded() {
        Logger.log(Log().withTag(TAG).text("loaded()"))
        listener?.onPageLoaded()
    }

    @JavascriptInterface
    fun handleCancel() {
        Logger.log(Log().withTag(TAG).text("handleCancel()"))
        listener?.onPageCancel()
    }

    @JavascriptInterface
    fun handleResult(result: String) {
        Logger.log(Log().withTag(TAG).text("handleResult(\"$result\")"))
        listener?.onPageResult(result)
    }

    @JavascriptInterface
    fun handleClose() {
        Logger.log(Log().withTag(TAG).text("handleClose()"))
        listener?.onPageClosed()
    }

    fun setListener(listener: EcosystemWebPageListener) {
        this.listener = listener
    }

    companion object {

        private val TAG = EcosystemNativeApi::class.java.simpleName
    }
}
